### PR TITLE
Tmp add liq spec

### PR DESCRIPTION
**Pull request by daejunpark**
Do not merge. Only for discussion purpose.

### CODE REVIEW

File: tests/proofs/dexter/dexter-spec.md [hunk 1]
```diff
@@ -15,6 +15,89 @@ module DEXTER-SPEC
...
+        <knownaddrs> KnownAddresses </knownaddrs>
+        <paramtype> #Type(#DexterVersionSpecificParamType(IsFA2)) </paramtype>
+        <operations> _
+                  => [ Transfer_tokens Pair Sender Pair SelfAddress ((Amount *Int TokenAmount) /Int XtzAmount) #Mutez(0) TokenAddress Nonce ] ;;
+                     [ Transfer_tokens Pair ((Amount *Int OldLqt) /Int XtzAmount) Owner #Mutez(0) LqtAddress (Nonce +Int 1) ] ;;
+                     .InternalList
+        </operations>
+    requires notBool IsFA2 // TODO Handle IsFA2==true
+     andBool CurrentTime <Int Deadline
+     andBool Amount <=Int MaxTokensDeposited
```
Comment (daejunpark) [inline tests/proofs/dexter/dexter-spec.md:85]: No such requirement exists in the code.

File: tests/proofs/dexter/dexter-spec.md [hunk 2]
```diff
@@ -15,6 +15,89 @@ module DEXTER-SPEC
...
+     andBool Amount      >Int 0
+     andBool XtzAmount   >Int 0
+     andBool OldLqt      >Int 0
+     andBool TokenAmount >Int 0
```
Comment (daejunpark) [inline tests/proofs/dexter/dexter-spec.md:89]: I know these are desired conditions, but the minimum requirement is only `XtzAmount   >Int 0`, right?

File: tests/proofs/dexter/dexter-spec.md [hunk 3]
```diff
@@ -15,6 +15,89 @@ module DEXTER-SPEC
...
+     andBool #mulDiv(Amount, TokenAmount, XtzAmount) <=Int MaxTokensDeposited
+     andBool (Amount *Int TokenAmount) %Int XtzAmount ==Int 0
+     /* TODO: Also handle this case:
+     andBool #mulDiv(Amount, TokenAmount, XtzAmount) <Int MaxTokensDeposited
```
Comment (daejunpark) [inline tests/proofs/dexter/dexter-spec.md:93]: Why `<Int` instead of `<=Int`?

File: tests/proofs/dexter/dexter-spec.md [hunk 4]
```diff
@@ -15,6 +15,89 @@ module DEXTER-SPEC
 
   claim <k> (now 0 => .) ~> #dexterCode(_) ... </k>
         <mynow> #Timestamp(0) </mynow>
+
+  claim <k> PUSH nat X:Int ; PUSH nat Y:Int ; PUSH nat Z:Int ; MUL ; EDIV ; IF_NONE {} { DUP ; CDR ; SWAP ; CAR ; PUSH nat 0 ; DIG 2 ; COMPARE ; EQ } => . ... </k>
+        <stack> .Stack => [ bool ?_ ] ; [ nat (Z *Int Y) /Int X ] ; .Stack </stack>
+    requires X >Int 0
+     andBool Y >=Int 0
+     andBool Z >=Int 0
```
Comment (daejunpark) [inline tests/proofs/dexter/dexter-spec.md:23]: Could you please explain what it is about?

File: tests/proofs/dexter/dexter-spec.md (hunk 2 above)
Comment (hjorthjort) [reply tests/proofs/dexter/dexter-spec.md:89]: Hmm, you're probably right. These simplified reasoning (this spec has been work in progress for over a week), but I guess we should be fine with just `XtzAmount > 0`. I'll see if I can generalize.

File: tests/proofs/dexter/dexter-spec.md (hunk 3 above)
Comment (hjorthjort) [reply tests/proofs/dexter/dexter-spec.md:93]: Note that this is commented out. If you look at this line and the line below, you see these would cover the various cases of `ceildiv(Amount*TokenAmout, XtzAmount)`. Lines 90 and 91 handle the case where the division has no remainder so no rounding happens, and lines 93 and 94 would cover the case when the remainder was positive, and thus the division would be rounded up.

File: tests/proofs/dexter/dexter-spec.md (hunk 4 above)
Comment (hjorthjort) [reply tests/proofs/dexter/dexter-spec.md:23]: This is a test case for the failure of Z3 on non-linear reasoning. This failed without the rewrites to `#mulMod` and `#mulDiv`.

File: tests/proofs/dexter/dexter-spec.md (hunk 1 above)
Comment (hjorthjort) [reply tests/proofs/dexter/dexter-spec.md:85]: Correct, this isn't necessary

File: tests/proofs/dexter/dexter-spec.md [hunk 5]
```diff
@@ -15,6 +15,80 @@ module DEXTER-SPEC
...
+-   Operations:
+
+1. call to `transfer` entrypoint of liquidity address: Send tokens from sender to self.
+2. call to `mintOrBurn` entrypoint of token address: Adds liquidity for the sender.
+
+-   Preconditions
+
+1.  the token pool is _not_ currently updating (i.e. `storage.selfIsUpdatingTokenPool = false`)
+2.  the deadline has not passed (i.e. the `Tezos.now < input.deadline`)
+3.  the tez transferred is less than `input.maxTokensDeposited`
```
Comment (daejunpark) [inline tests/proofs/dexter/dexter-spec.md:53]: it should be "token" transferred, not "tez" transferred here.

**Comment by daejunpark** [comment]
The spec looks great, thank you!

File: tests/proofs/dexter/dexter-spec.md (hunk 5 above)
Comment (hjorthjort) [reply tests/proofs/dexter/dexter-spec.md:53]: thanks, thought I fixed that, must have undone accidentally

**Pull request closed without merging**